docs/vm/access-to-the-vm.md

```diff
@@ -19,13 +19,55 @@ Additionally, you can connect directly from your computer's SSH client.
 
 VMs can be accessed from the UI directly using either VNC or the serial console.
 
-If the VGA display is not enabled on the VM, as with the Ubuntu minimal cloud image, the VM can be accessed with the serial console.
+If the VGA display is not enabled on the VM, as with the `Ubuntu-minimal-cloud` image, the VM can only be accessed with the serial console.
 
 ![](assets/access-to-vm.png)
 
-## Access with the SSH Client
+## SSH Access
 
-Enter the IP address of the host in a terminal emulation client, such as PuTTY. You may also run the following command to access the VM directly from your computer's SSH client:
+Harvester provides two ways to inject SSH public keys into virtual machines. Generally, these methods fall into two categories. [Static key injection](#static-ssh-key-injection-via-cloud-init), which places keys on the cloud-init when the virtual machine is first powered on; [dynamic injection](#dynamic-ssh-key-injection-via-qemu-guest-agent), which allows keys or basic auth to be updated dynamically at runtime.
+
+### Static SSH Key Injection via Cloud Init
+
+You can provide ssh keys to your virtual machines during the creation time on the `Basics` tab. Additionally, you can place the public ssh keys into your cloud-init script to allow it to take place.
+
+![](assets/vm-ssh-keys.png)
+
+#### Example of SSH key cloud-init configuration:
+```yaml
+#cloud-config
+ssh_authorized_keys:
+  - >-
+    ssh-rsa #replace with your public key
+```
+
+
+### Dynamic SSH Key Injection via Qemu guest agent
+
+_Available as of v1.0.1_
+
+Harvester supports dynamically injecting public ssh keys at run time through the use of the qemu guest agent. This is achieved through the `qemuGuestAgent` propagation method.
+
+!!! note
+    This method requires the qemu guest agent to be installed within the guest VM.
```

Comment:
We should consider linking to the installation doc, if any, here.
```suggestion
    This method requires the qemu guest agent to be installed within the guest VM.
```

Reply:
There is a built-in UI option so maybe is not needed, thanks,